lms/djangoapps/discussion/rest_api/serializers.py

```diff
@@ -425,11 +427,37 @@ def update(self, instance, validated_data):
             requesting_user_id = self.context["cc_requester"]["id"]
             if key == "closed" and val:
                 instance["closing_user_id"] = requesting_user_id
+                self.trigger_thread_closed_event(instance, validated_data)
             if key == "body" and val:
                 instance["editing_user_id"] = requesting_user_id
         instance.save()
         return instance
 
+    def trigger_thread_closed_event(self, instance, validated_data):
+        """
+        Triggers edx.forum.thread.locked tracking event
+        """
+        url_kwargs = {
+            'course_id': instance.course_id,
+            'discussion_id': instance.commentable_id,
+            'thread_id': instance.id
+        }
+        user = self.context['request'].user
+        user_course_roles = [role.role for role in user.courseaccessrole_set.filter(course_id=instance.course_id)]
+        user_forums_roles = [role.name for role in user.roles.filter(course_id=instance.course_id)]
+        tracker.emit(
+            'edx.forum.thread.locked',
+            {
+                'id': instance.id,
+                'team_id': get_team(instance.commentable_id),
+                'url': reverse('single_thread', kwargs=url_kwargs),
```

Comment:
Will this url be useful? It is only for the api and cannot be accessed. Wondering why this is useful then?